daft/udf.py

```diff
@@ -394,7 +394,7 @@ def __hash__(self) -> int:
 
 def udf(
     *,
-    return_dtype: DataType,
+    return_dtype: DataType | type,
```

Comment:
I wonder if we could create a type alias in pythong for `DataType | type` as it seems to be used in a lot of places

Reply:
I remember documentation being a bit messy when I used type aliases for something else. Let me give it a shot again and see how it shows up on docs.

Reply:
ah I forgot how it shows up in documentation... yea I guess I am thinking like a Rust user. it might be complicated for something that isn't statically typed having to go deeper in docs 🤔. Trust your judgement here.